examples/sn-dms-demo/src/components/Shared.tsx

```diff
@@ -1,16 +1,185 @@
+import ActionMenu from './ActionMenu/ActionMenu'
+import { DisplayNameCell } from './ContentList/CellTemplates/DisplayNameCell'
+import { RenameCell } from './ContentList/CellTemplates/RenameCell'
+import { DefaultCell } from './ContentList/CellTemplates/DefaultCell'
+import { DateCell } from './ContentList/CellTemplates/DateCell'
+import { ReferencedUserCell } from './ContentList/CellTemplates/ReferencedUserCell'
 
-export const Shared: React.StatelessComponent = () => {
-  return (
-    <AppBar position="static" style={{ background: '#fff' }}>
-      <Toolbar style={{ display: 'flex', flexDirection: 'row', padding: '0 12px' }}>
-        <div style={{ flex: 1, display: 'flex' }}>
+const mapStateToProps = (state: rootStateType) => ({
+  sharedItems: state.dms.shared.sharedItems,
+  selected: state.dms.shared.selected,
+  active: state.dms.shared.active,
+  docLib: state.dms.documentLibrary.parentIdOrPath,
+  hostName: state.sensenet.session.repository ? state.sensenet.session.repository.repositoryUrl : '',
+  editedItemId: state.dms.editedItemId,
+  currentUser: state.sensenet.session.user.content,
+})
+
+const mapDispatchToProps = {
+  getSharedItems,
+  select,
+  setActive,
+  openActionMenu,
+  closeActionMenu,
+  updateContent,
+  resetSearchValues,
+  updateChildrenOptions,
+}
+
+class Shared extends React.Component<
+  ReturnType<typeof mapStateToProps> & typeof mapDispatchToProps & RouteComponentProps
+> {
+  private load() {
+    const user = this.props.currentUser
+    this.props.getSharedItems(user)
+  }
+
+  public componentDidMount() {
+    this.load()
+  }
+
+  private handleOpenItem(content: GenericContent) {
+    if (content.IsFolder) {
+      const newPath = `${this.props.match.path.substr(0, this.props.match.path.lastIndexOf('/'))}/${btoa(
+        content.Path.substr(0, content.Path.lastIndexOf('/')),
+      )}`
+      this.props.history.push(newPath)
+    } else {
+      const newPath = `${this.props.match.path.substr(0, this.props.match.path.lastIndexOf('/'))}/${btoa(
+        content.Path.substr(0, content.Path.lastIndexOf('/')),
+      )}/preview/${btoa(content.Id as any)}`
+      this.props.history.push(newPath)
+    }
+  }
+
+  public render() {
+    return (
+      <div style={{ padding: '2em' }}>
+        <MuiThemeProvider theme={contentListTheme}>
           <Typography variant="h5">Shared</Typography>
-        </div>
-      </Toolbar>
-    </AppBar>
-  )
+          <ContentList
+            schema={{ FieldSettings: [] } as any}
+            selected={this.props.selected}
+            onRequestSelectionChange={this.props.select}
+            active={this.props.active}
+            icons={icons}
+            items={this.props.sharedItems}
+            fieldsToDisplay={['DisplayName', 'ModifiedBy', 'ModificationDate']}
+            onRequestActiveItemChange={this.props.setActive}
+            displayRowCheckbox={false}
+            onRequestActionsMenu={(ev, content) => {
+              ev.preventDefault()
```

Comment:
I see an error when I try to open context menu in shared view. I've logged in with devdog and businesscat shared a doc with me. `Uncaught TypeError: content.Actions.filter is not a function or its return value is not iterable`